lint-requirements.txt

```diff
@@ -1 +1,2 @@
+clang-format>=18.1.5
```

Comment:
This is the version that the current clang-format config in yscope-dev-utils was written for. We should eventually add a requirements.txt file to yscope-dev-utils so this will be clearer.

Reply:
Looking at https://github.com/y-scope/clp/issues/545 and https://github.com/y-scope/clp/pull/546 , shall we write `~=` instead?
```suggestion
# Lock to v18.x until we can upgrade our code to meet v19's formatting standards.
clang-format~=18.1
```